src/window-rules.c

```diff
@@ -55,10 +30,15 @@ other_instances_exist(struct window_rule *rule, struct view *self)
 static bool
 view_matches_criteria(struct window_rule *rule, struct view *view)
 {
-	if (rule->match_once && other_instances_exist(rule, view)) {
+	struct view_query query;
```

Comment:
Suggest `struct view_query query = {0};`, otherwise stuff might be uninitialized and when we add further conditions in the future this might break horribly.

Reply:
Just so don't do something dumb: This is the definition of `struct view_query`:

```c
struct view_query {
	struct wl_list link;
	char *identifier;
	char *title;
	int window_type;
};
```

`link` is not used in `view_matches_query()`. Do we have to initialize it anyway? Would that work with both options you proposed? (I am not that experienced with C)

Reply:
Yes, AFAIR that should always work.
Everything not defined within the initializer is set to `0` or `0 equivalent` (e.g. `NULL` pointer).

CC @ahesford to ensure I don't talk nonsense.

Reply:
Yes, if you initialize with ` = { something, ... }` and the list of specified values is not complete, every other field in the struct will be default initialized (which would be `0` in these cases).

If you don't initialize at all, then static variables are default initialized but local variables will be uninitialized, inheriting whatever garbage was on the stack (or heap) at the time they are created.